### PR TITLE
gosimports: init at 0.3.8

### PR DESCRIPTION
**Pull request by maolonglong**
## Description of changes

<https://github.com/rinchsan/gosimports>

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/gosimports/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  subPackages = [ "cmd/gosimports" ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.version=${version}"
+  ];
+
+  meta = with lib; {
+    homepage = "https://github.com/rinchsan/gosimports";
```
Comment (mfrw) [inline pkgs/development/tools/gosimports/default.nix:28]: Can we please add a:
`mainProgram = "gosimports";`

File: pkgs/development/tools/gosimports/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+buildGoModule rec {
+  pname = "gosimports";
+  version = "0.3.8";
+
+  src = fetchFromGitHub {
+    owner = "rinchsan";
+    repo = "gosimports";
+    rev = "v${version}";
+    sha256 = "sha256-xM1CGW8UB+VHN+2Rm6cF/1bOBVDeUG+6kxUxUcvP7FM=";
```
Comment (MikaelFangel) [inline pkgs/development/tools/gosimports/default.nix:14]:
```suggestion
    hash = "sha256-xM1CGW8UB+VHN+2Rm6cF/1bOBVDeUG+6kxUxUcvP7FM=";
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -28434,6 +28434,8 @@ with pkgs;
 
   gotools = callPackage ../development/tools/gotools { };
 
+  gosimports = callPackage ../development/tools/gosimports { };
```
Comment (MikaelFangel) [inline pkgs/top-level/all-packages.nix:28437]:
```suggestion
```

The package should be placed on the new `pkgs/by-name/so/some-example/package.nix` hierarchy 😊

Comment (maolonglong) [reply pkgs/top-level/all-packages.nix:28437]: Oh, I just noticed this folder. This hierarchy is really a good idea.

**Comment by mfrw** [comment]
Result of `nixpkgs-review pr 268867` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gosimports</li>
  </ul>
</details>

**Review by kirillrdy — APPROVED** [review]
Result of `nixpkgs-review pr 268867` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>gosimports</li>
  </ul>
</details>

**Review by mfrw — APPROVED** [review]

**Review by MikaelFangel — APPROVED** [review]

File: maintainers/maintainer-list.nix [hunk 4]
```diff
@@ -11008,6 +11008,12 @@
     githubId = 3507;
     name = "Michael Fellinger";
   };
+  maolonglong = {
+    email = "shaolong.chen@outlook.it";
+    github = "maolonglong";
+    githubId = 50797868;
```
Comment (fabianhjr) [inline maintainers/maintainer-list.nix:11014]: Matched: https://api.github.com/users/maolonglong

**Comment by fabianhjr** [comment]
Thanks for contributing to nixpkgs! :3

**Pull request merged**